### PR TITLE
Add versioning notes to release docs

### PR DESCRIPTION
**Pull request by aaronlevy**
@diegs @yifan-gu

### CODE REVIEW

File: RELEASING.md [hunk 1]
```diff
@@ -1,5 +1,35 @@
...
+### Checkpointer Versioning
+
+The checkpointer is developed in the same repo, but can be thought of as an independent project.
+Because of this we do not use tags for the checkpointer releases, as they would intermix with bootkube releases (which do not coincide).
+
+Instead, checkpointers are released using the last git-hash of the changes added to the checkpointer subtree.
+Available releases can be seen on the Quay repository: https://quay.io/repository/coreos/pod-checkpointer.
+If there were no changes made to the checkpointer subtree, a new release is not necessary.
+
+Eventually we might want to consider moving the checkpointer to it's own repo.
```
Comment (diegs) [inline RELEASING.md]: nit: s/it's/its

Comment (diegs) [inline RELEASING.md]: maybe also link to some of the upstream checkpointer discussions?

https://github.com/kubernetes/kubeadm/issues/131
https://github.com/kubernetes/kubernetes/issues/489
https://github.com/kubernetes-incubator/bootkube/issues/424

Comment (aaronlevy) [inline RELEASING.md]: Added a bit more wording here

**Comment by aaronlevy** [comment]
@diegs ready for another pass

**Pull request merged**